### DISCUSSION TITLE
0.5.1

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#319] [Announcements]
If this is a general bug report, please check for existing reported issues, or create a new issue.

Documentation: https://kubilus1.github.io/autoortho/

Change of note include:
* Re-worked download mechanism that should provide retries and better file verification.
* Optional installer for Windows users replaces the stand-alone all-in-one exe release.  The packaged zip release is still available.
* More flexible management of scenery packages which should allow handling different scenery package types going forward.

Also note:  This upgrade may require a re-setup and re-download of scenery packages, particularly if you are upgrading from 0.4.x or older versions.

## What's Changed
* Update certs by @kubilus1 in https://github.com/kubilus1/autoortho/pull/228
* Rework download mechanism by @kubilus1 in https://github.com/kubilus1/autoortho/pull/264
* Speed up fuse tear-down by @jonaseberle in https://github.com/kubilus1/autoortho/pull/268
* use refuse in test by @jonaseberle in https://github.com/kubilus1/autoortho/pull/256
* Update WinFSP section to work on Windows 11 by @206airmail in https://github.com/kubilus1/autoortho/pull/286
* Create installer for Windows systems by @kubilus1 in https://github.com/kubilus1/autoortho/pull/272
* adjust image referenced in aoimage test by @jonaseberle in https://github.com/kubilus1/autoortho/pull/267

## New Contributors
* @206airmail made their first contribution in https://github.com/kubilus1/autoortho/pull/286

**Full Changelog**: https://github.com/kubilus1/autoortho/compare/0.5.0...0.5.1

**Comment by MacMaxYT** [comment]
Hello, 
i'm getting errors like: 
`WARNING:getortho:Failed to get chunk Chunk(34445,21664,BI,16,12) on server a. Err: timed out URL: http://r1.ortho.tiles.virtualearth.net/tiles/a1202031030201101.jpeg?g=136`
`WARNING:getortho:Failed getting: Chunk(34445,21664,BI,16,12) () {}, re-submit.`

before the update i was also getting:
```
WARNING:getortho:Failed to get chunk Chunk(4820,3381,BI,13,9) on server c. Err:
WARNING:getortho:Failed getting: Chunk(4820,3381,BI,13,9) () {}, re-submit.
```

What can be the problem? Before the update VPN was solution but now i'm getting same even with it

**Reply by kubilus1** [reply]
Timeouts will happen on occasion, there is nothing really new with this code path.  What you are showing is a warning, not an error.  Timeouts will be due to your path to whatever bing server you end being routed to.

**Reply by MacMaxYT** [reply]
Okay i understand, but I'm always getting many many empty, just green, chunks like this -> https://imgur.com/DILMcft

Those green squares make me crazy

**Reply by kubilus1** [reply]
Interesting.  Is this on Windows?  Can you take a look at your running processes and see if 'Antimalware Executable' is taking a lot of CPU?  I think Windows Defender run-time protection really slows stuff down on Windows. You may have luck making an exception to the your cache dir and download dir locations.  (By default 'C:\Users\<youruser>\.autoortho-data')

You can also tweak the 'max_wait' time in your config, if things are still slow for you.  (By default it's 0.5 seconds, but try increasing to 1 then maybe 2 and see if this helps)

**Reply by MacMaxYT** [reply]
I was always having this issue and i done everything i found on reddit github etc. 

Yes, it’s windows 10, but i was having same on windows 11
It’s x-plane 11 but i had same in x-plane 12 

I disabled windows defender real time protection. 
I disabled windows firewall 
I also change the max wait time to 2.

I done everything on previous version, 0.4.
Only solution for me was using VPN all during the flight (i guess my ISP is blocking something)
But now in 0.51 beta even this won’t help 

And it’s also important not every VPN was working, when i used NordVPN i got same error, same cyberghost. Only what was working was FastVPN from namecheap (i guess they didn’t have such protections)

**Reply by MacMaxYT** [reply]
do you have any idea what can fix it?

**Reply by MacMaxYT** [reply]
https://imgur.com/a/0fPgxC9 - console log

https://imgur.com/a/YTOMdUA - How my game looks like

This is 0.4.4 but in 0.5.1 it's SAME

as I told, real-time protection off, firewall off, maxwait = 2

**Reply by MacMaxYT** [reply]
I downloaded also 0.5.1 now, here is the effect:

Console: 
![image](https://github.com/kubilus1/autoortho/assets/71557093/192e5575-6596-4db4-b833-5ed7e228d701)

Effect in game is same as 0.4.4

**Reply by MacMaxYT** [reply]
![image](https://github.com/kubilus1/autoortho/assets/71557093/ed956e4c-2fab-4d95-b4a7-369162378de8)

And thats not once, this happens always and everywhere

**Reply by darkpino** [reply]
If you ever find a solution i'd love to hear it as I have the same problem unfortunately

**Comment by sergiolamares** [comment]
Hi! how do I compile this? what do I need more, besides Python? thanks

**Reply by kubilus1** [reply]
It's python so it's not compiled.  General setup instructions are here: https://kubilus1.github.io/autoortho/ most users can just use the packaged release.

If you choose to run via source, besides the pre-reqs listed in the instructions, just perform standard pip install of 'requirements.txt'.

**Reply by sergiolamares** [reply]
but is there a compiled package of 0.5.1 ? I can't run 0.5.0 it always gives error for missing DDS files

**Reply by kubilus1** [reply]
https://github.com/kubilus1/autoortho/releases

**Reply by sergiolamares** [reply]
Oh the EXE wasn't avaiable yesterday! I'll try it... thanks

**Reply by kubilus1** [reply]
Yeah, there can be a bit of a delay between the creation of the release and the builds to happen.  I also had to rebuild this once.

**Comment by sergiolamares** [comment]
No worries! Thanks.
I want to install the packges, intended to be on X-Plane custom Scenery folder, on a different drive. From there witch AutoOrtho folders need to be linked on custom scenery?
Thanks

Enviado do meu iPhone

No dia 18/08/2023, às 18:15, kubilus1 ***@***.***> escreveu:

﻿

Yeah, there can be a bit of a delay between the creation of the release and the builds to happen. I also had to rebuild this once.

**Reply by kubilus1** [reply]
So basically you can setup and run the program which will create the mount points like 'z_ao_na'  in the 'Custom Scenery' folder you create. 

On Linux you can just symlink each to XPlane's actual custom scenery dir like any other package.  In Windows you can just make a short cut and put that in the same location.

**Comment by sergiolamares** [comment]
That’s the problem, it doesn’t create nothing, no points at all… or I set something wrong… even setting custom scenery to XP’s custom scenery folder, it doesn’t create any points, links or junctions…

Enviado do meu iPhone

No dia 18/08/2023, às 19:27, kubilus1 ***@***.***> escreveu:

﻿

So basically you can setup and run the program which will create the mount points like 'z_ao_na' in the 'Custom Scenery' folder you create.

On Linux you can just symlink each to XPlane's actual custom scenery dir like any other package. In Windows you can just make a short cut and put that in the same location.

**Reply by kubilus1** [reply]
You have to install scenery first.  The log messages should point you to what is going on.

**Reply by sergiolamares** [reply]
I've installed EUR package, and I've reconfirmed, it creates a z_autoortho folder but it doesn't match XP custom scenery file structure:
it did create: z_autoortho > scenery > z_ao_eur... no junction at all... in the end the issue persists

0:01:31.417 E/SYS: +-------------------------------------------------------------------------------
0:01:31.417 E/SYS: | Error, could not locate image file for terrain
0:01:31.417 E/SYS: | Custom Scenery/z_ao_eur/terrain/24832_31136_BI16_sea_overlay.ter
0:01:31.417 E/SYS: | Custom Scenery/z_ao_eur/terrain/
0:01:31.417 E/SYS: | ../textures/24832_31136_BI16.dds
0:01:31.417 E/SYS: | (REN_terrain.cpp:206)
0:01:31.417 E/SYS: +-------------------------------------------------------------------------------

Any help please?

**Reply by kubilus1** [reply]
That's the Xplane log, I would need the relevant lines from autoortho's logs.  C:\Users\youuser\.autoorth-data\logs

**Reply by kubilus1** [reply]
Also what OS is this, specifically?

**Reply by sergiolamares** [reply]
[autoortho.log](https://github.com/kubilus1/autoortho/files/12383992/autoortho.log)
Here it is.
Windows 11 Pro

**Reply by kubilus1** [reply]
I fail to see any issues in that log.  Maybe try Bing maps instead of Firefly.  Also try uninstalling Dokan (reboot after uninstall!) and install WinFSP.  Some people on Windows 11 have better luck with WinFSP.

**Reply by sergiolamares** [reply]
I will. I also resintalled everything on a different drive, can you point me the path of folders to be linked in XP Custom Scenery folder?

**Reply by kubilus1** [reply]
Take a look at the documentation, which should walk you through the steps.  https://kubilus1.github.io/autoortho/

**Reply by sergiolamares** [reply]
Finaly I understood how things work with AO, it needs to be running, to create mounts/links, I hadn't sort that out... 
It's working now, even with the Firefly maps! 
Thanks!

**Reply by kubilus1** [reply]
Great!!

**Reply by sergiolamares** [reply]
Yes it was really driving me nuts!
Now, is there a way to remove the anoying warning about sencery packages not beeing installed in X-Plane Costum Scenery folder? I can prove it works perfectly fine installing them on a different drive.
Please consider that option in a future update...
Thanks!

**Reply by kubilus1** [reply]
@sergiolamares When you got this working were you setup with Dokan or WinFSP?  I like to understand what combinations work for people.  Thanks!

**Reply by sergiolamares** [reply]
I got it working with WinFSP, per your suggestion, of unstalling Dokan

**Comment by flyman-dc8** [comment]
![image](https://github.com/kubilus1/autoortho/assets/141371015/79db45cd-e9c4-4af1-b87c-fd32a6c5f2d4)
[autoortho.zip](https://github.com/kubilus1/autoortho/files/12383682/autoortho.zip)
Still same issue 
![258246185-fe5f09d4-4c96-496b-b36a-32945219cd7f](https://github.com/kubilus1/autoortho/assets/141371015/f9fe6924-bf5a-4712-a405-45086616c823)

**Reply by kubilus1** [reply]
Running from source I see.  Did you try to just pip install the missing 'packaging' package?

**Reply by kubilus1** [reply]
Is there a reason you aren't using the built zip or bin?

**Reply by flyman-dc8** [reply]
I will the built zip. 
![image](https://github.com/kubilus1/autoortho/assets/141371015/bbf373ca-1101-4844-9152-54a45e4d638e)

Installed now

**Reply by flyman-dc8** [reply]
Nope still not working I did try both the zip and the exe
[autoortho.zip](https://github.com/kubilus1/autoortho/files/12384899/autoortho.zip)

![image](https://github.com/kubilus1/autoortho/assets/141371015/305b5873-3a82-460e-9d27-424c28576f48)

**Reply by rquintelarj** [reply]
Same to

SBKP Scenary dont open with 0.5.0 version.

**Reply by kubilus1** [reply]
Something is up with logging.  Fixing this and replacing the bins.  Once these are done, retry and send the updated log and I'll take a look.

**Reply by DemonTraitor** [reply]
I am getting the same since updating... all installed fine. xOrganizer is also moaning about the scenery.

Log file is empty.

**Reply by flyman-dc8** [reply]
[autoortho.log](https://github.com/kubilus1/autoortho/files/12386685/autoortho.log)
 
Installed new version still same error

**Reply by kubilus1** [reply]
Yeah I've never seen that error before.  Immediately it looks like whatever is received is corrupt.   Can you try using Dokan instead of WinFSP.  Also maybe make sure WinFSP is the latest version.

**Reply by flyman-dc8** [reply]
Ok, I will try that. On another note I work in IT on the other side Infrastructure and I know how much time it takes developers to work on the backend stuff. I can't tell thank you enough for how much help you are putting into this and with that It's all free. 
 Thank you for all the help.

**Reply by flyman-dc8** [reply]
Also, do you think it can be something with the x86 or x64 of Dokan? Should I try the X86 I am on Windows 11

**Reply by flyman-dc8** [reply]
Doken x64 Logfile same error
[autoortho.log](https://github.com/kubilus1/autoortho/files/12386749/autoortho.log)

**Reply by flyman-dc8** [reply]
In your setup documentation you say 
"You have setup and configured [Dokan](https://github.com/dokan-dev/dokany/releases/latest)"
I installed it do i need to configure anything with in the software?

**Reply by kubilus1** [reply]
Thanks!  I mostly do infrastructure as well, at least recently.

Mostly I use Linux, so I'm less familiar with Windows, but I think with Dokan there isn't anything to configure and you install the 64 bit version.

With your error, is it possible your ISP is blocking things, or maybe you are IP is blacklisted?  Maybe try one of the other map types, such as firefly.

How familiar are you with Python?   If you can, install pytest, checkout the source code, install all the reqs, go into the 'autoortho' dir and run the tests, particularly:  `pytest -v test_getortho.py`

**Reply by flyman-dc8** [reply]
I don't know much about Python but i will give it a try. 

I did get this error in xorganizer

Issues that need attention are listed below. Please review them and apply a fix if needed.
Some issues are reported only once and further on ignored. With a complete rescan they are reported again.
If there are any bad links found (= symlinks, junctions or shortcuts), just delete and recreate them.
Lines: # = Serious error | ! = Needs Attention | - = For your information.
════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════

# Error reading folder or link          : E:\X-Plane 12\
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Aug 19 2023 21:12:41 | Scan of the custom scenery folder
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
# Error reading folder or link          : E:\X-Plane-12\Custom Scenery\z_ao_na

**Reply by Tiggs62** [reply]
Same issue. Win 11, XP12.06b5. Was running 0.4.4 ok, was running 0.5.0 (but seeing quite a few green squares) but 0.5.1 (fresh install, scenery reloaded) is giving me the same issue as here:
![image](https://github.com/kubilus1/autoortho/assets/54512432/fbcfa39f-0c4a-499d-9edf-ffb13422b275)

z_ao-eur is present in my custom scenry folder and listed in the scenery.ini

Curiously its a shortcut, and if I try to go to it I get a message saying it is not accessible or unavailable.

**Reply by kubilus1** [reply]
Are you certain AutoOrtho is running?  If it says unavailable that tells me it probably isn't.  Also XPlane screenshots are not particularly useful for diagnosing issues, but errors from autoortho itself or log snippets may be.

**Reply by Tiggs62** [reply]
Definitely running... but I think what it was doing was trying to run an old version still partly installed. I Did Add/Remove and removed autoortho entirely, then reinstalled 0.5.1 and its working fine! Thank you.

**Reply by flyman-dc8** [reply]
So I uninstalled everything and ran ccleaner to clean the registry. Then I reinstalled AO and still got the same error. Log attached.
[autoortho.log](https://github.com/kubilus1/autoortho/files/12389970/autoortho.log)

**Reply by flyman-dc8** [reply]
I wonder if WinFSP cant set the spaces in the path when it mounts the sub folder. Like in the BOLD below
"INFO:autoortho:AutoOrtho:  root: E:/**X-Plane 12/Custom Scenery**\z_autoortho\scenery\z_ao_na  mountpoint: E:/**X-Plane 12/Custom Scenery**\z_ao_na"

**Reply by flyman-dc8** [reply]
(https://learn.microsoft.com/en-us/troubleshoot/windows-server/deployment/filenames-with-spaces-require-quotation-mark)

**Comment by TheSpizzo** [comment]
Where is this fucking .exe ?

**Reply by dpdurst** [reply]
Honestly that language is uncalled for on here! Seriously get a grip..

**Reply by kubilus1** [reply]
People like these are why projects go dormant.  Another one blocked!

**Comment by coptered** [comment]
I am using WinFSP on Windows 11 and AO is creating the z_ao_na shortcut, but XP12.06b6 keeps moving it to the top of the scenery_packs.ini file.  Is there some way to lock it down in the scenery_packs.ini?
[Log.txt](https://github.com/kubilus1/autoortho/files/12403753/Log.txt)

**Reply by coptered** [reply]
I have added the shortcut to the scenery_packs.ini and with AutoOrtho running and the mount point available, Xplane 12 moves it to the top of the scenery_packs.ini. I have tried it in the Custom Scenery folder, and linked it to another drive folder, both ways with the same results of Xplane 12 moving it to the top of the scenery_packs.ini.

**Comment by geardownandlocked** [comment]
I deleted the old version and put the new version in and downloaded all the scenery but the Custom Scenery folder now has just two subfolders - one folder called yAutoOrtho_Overlays and the other z_autoortho. The latter has a subfolder called scenery and then that folder has all the z_ao_eur etc folders. Is this right? Having run X-plane 12, there are now short cuts to the z_ao_ eur etc folders in the Custom Scenery folder. These don't work unless you have autoortho running. The manual seems to imply those z_ao_eur etc folders are at the high level in Custom Scenery, not as a subfolder of the z_autoortho. I am obviously doing something wrong.

**Reply by jonaseberle** [reply]
No, sounds good. These are not shortcuts though, but mount points. They only work when the "driver" (AutoOrtho) is running. Make sure they are added to `scenery_packs.ini` (on the bottom).

**Reply by jonaseberle** [reply]
@coptered this is not your thread. I was trying to help a user here. Please stop hijacking.

**Comment by jmihalich** [comment]
So, i uninstalled the last version, and install the latest 0.5.1-3 on windows 11 into the default install folder.  The app won't even launch now and is logging this error:
`INFO:__main__:Setup logs: C:\Users\joe\.autoortho-data\logs, level: 20
INFO:autoortho:AutoOrtho version: 0.5.1-3
INFO:aostats:Creating stats object
WARNING:autoortho:No installed sceneries detected.  Exiting.`

what am i doing wrong?  obviously there's no sceneries installed, it's a fresh install.

**Comment by thewalkingfruit** [comment]
When updating from 0.4.4 to this version got a warning that Xplane cannot run with the package installed, uninstalled dokan, rebooting and installing winfsp makes it work, for anyone having the same issue, and for some feedback. Thanks for your work

**Reply by DemonTraitor** [reply]
When I use WinFSP, XP removed the scenery from the ini, then puts it back in the wrong order

**Comment by marloscaida** [comment]
please, see this explanation:
https://forums.x-plane.org/index.php?/forums/topic/295203-solved-60-1-51-3-custom-sceneryz_ao_na-x-plane-cannot-run/&page=2#comment-2618022
